src/main/java/com/google/cloud/tools/app/cloudsdk/CloudSdkAppEngineFactory.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.app.cloudsdk;
+
+import com.google.cloud.tools.app.api.AppEngineFactory;
+import com.google.cloud.tools.app.api.deploy.AppEngineDeployment;
+import com.google.cloud.tools.app.api.deploy.AppEngineFlexibleStaging;
+import com.google.cloud.tools.app.api.deploy.AppEngineStandardStaging;
+import com.google.cloud.tools.app.api.devserver.AppEngineDevServer;
+import com.google.cloud.tools.app.api.genconfig.GenConfigUtility;
+import com.google.cloud.tools.app.api.instances.AppEngineInstances;
+import com.google.cloud.tools.app.api.logs.AppEngineLogs;
+import com.google.cloud.tools.app.api.services.AppEngineServices;
+import com.google.cloud.tools.app.api.versions.AppEngineVersions;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineDeployment;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineDevServer;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineFlexibleStaging;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineGenConfig;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineInstances;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineLogs;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineServices;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineStandardStaging;
+import com.google.cloud.tools.app.cloudsdk.internal.CloudSdkAppEngineVersions;
+import com.google.cloud.tools.app.cloudsdk.internal.sdk.CloudSdk;
+import com.google.cloud.tools.app.cloudsdk.process.ProcessExitListener;
+import com.google.cloud.tools.app.cloudsdk.process.ProcessOutputLineListener;
+import com.google.cloud.tools.app.cloudsdk.process.ProcessStartListener;
+
+import java.io.File;
+
+/**
+ * A factory for creating Cloud SDK-based implementations of the App Engine services.
+ */
+public class CloudSdkAppEngineFactory implements AppEngineFactory {
+
+  private CloudSdk cloudSdk;
+
+  private CloudSdkAppEngineFactory(CloudSdk cloudSdk) {
+    this.cloudSdk = cloudSdk;
+  }
+
+  @Override
+  public AppEngineDeployment deployment() {
```

Comment:
If you think that there's value in exposing the new package structure
independently of this Factory change, I'd remove it and go ahead with the
package refactor.  Otherwise I'd work with Appu on a design that you can
agree on.

On Mon, Jun 6, 2016 at 5:09 PM, meltsufin <notifications@github.com> wrote:

> In
> src/main/java/com/google/cloud/tools/app/cloudsdk/CloudSdkAppEngineFactory.java
> <https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/pull/105#discussion_r65969601>
> :
>
> > +
> > +import java.io.File;
> > +
> > +/**
> > + * A factory for creating Cloud SDK-based implementations of the App Engine services.
> > + */
> > +public class CloudSdkAppEngineFactory implements AppEngineFactory {
> > +
> > +  private CloudSdk cloudSdk;
> > +
> > +  private CloudSdkAppEngineFactory(CloudSdk cloudSdk) {
> > +    this.cloudSdk = cloudSdk;
> > +  }
> > +
> > +  @Override
> > +  public AppEngineDeployment deployment() {
>
> The methods can be renamed to create..., but in general, should we keep
> the factory or revert to exposing the CloudSdk along with the
> CloudSdkAppEngine....(CloudSdk) classes instead?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/pull/105/files/5723a07a6bb84c29656eee4b39a1bb8a36e1b524#r65969601>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHf5HTUZtbudsipTFfx9e804715g96X9ks5qJIyQgaJpZM4IvDJS>
> .
>